gensim/models/sent2vec.py

```diff
@@ -0,0 +1,307 @@
+        self.lr = lr
+        self.lr_update_rate = lr_update_rate
+        self.epochs = epochs
+        self.min_count = min_count
+        self.neg = neg
+        self.word_ngrams = word_ngrams
+        self.loss = loss
+        self.bucket = bucket
+        self.t = t
+        self.minn = minn
+        self.maxn = maxn
+        self.dropoutk = dropoutk
+
+    def train(self, sentences):
+        print "Creating dictionary..."
+        self.dict = Dictionary(t=self.t, bucket=self.bucket, maxn=self.maxn, minn=self.minn)
+        self.dict.read(sentences=sentences, min_count=self.min_count)
+        print "Dictionary created, dictionary size:", self.dict.size, ",tokens read:", self.dict.ntokens
+        self.token_count = 0
+        counts = [entry.count for entry in self.dict.words]
+        print "Initializing model..."
+        self.model = Model(vector_size=self.vector_size, dict_size=self.dict.size, neg=self.neg, bucket=self.bucket)
+        self.model.init_table_negatives(counts=counts)
+        ntokens = self.dict.ntokens
+        local_token_count = 0
+        print "Training..."
+        progress = 0
+        start_time = time.time()
+        for i in range(self.epochs):
+            print "Begin epoch", i, ":"
+            for sentence in sentences:
+                progress = self.token_count / (self.epochs * ntokens)
+                if progress >= 1:
+                    break
+                lr = self.lr * (1.0 - progress)
+                ntokens_temp, hashes, words = self.dict.get_line(sentence)
+                local_token_count += ntokens_temp
+                if len(words) > 1:
+                    for i in range(len(words)):
+                        if random.uniform(0, 1) > self.dict.pdiscard[words[i]]:
+                            continue
+                        context = list(words)
+                        context[i] = 0
+                        context = self.dict.add_ngrams_train(context=context, n=self.word_ngrams, k=self.dropoutk)
+                        self.model.update(input_=context, target=words[i], lr=lr)
+                if local_token_count > self.lr_update_rate:
+                    self.token_count += local_token_count
+                    local_token_count = 0
+                if self.token_count >= self.epochs * ntokens:
+                    break
+            print "Progress: ", progress * 100, "% lr: ", lr, " loss: ", self.model.loss / self.model.nexamples
+        print "\n\nTotal training time: %s seconds" % (time.time() - start_time)
+
+    def sentence_vectors(self, sentence_string):
+        sentence = tokenize(sentence_string)
+        ntokens_temp, hashes, words = self.dict.get_line(sentence)
+        sent_vec = np.zeros(self.vector_size)
+        line = self.dict.add_ngrams(context=words, n=self.word_ngrams)
+        for word_vec in line:
+            sent_vec += self.model.wi[word_vec]
+        sent_vec *= (1.0 / len(line))
+        return sent_vec
+
+    def similarity(self, sent1, sent2):
```

Comment:
sent1 and sent2 should be **already** list of tokens (no need to tokenize it).

Reply:
Done. Kindly verify in the current commit.